src-input/duk_numconv.c

```diff
@@ -91,16 +91,40 @@ DUK_LOCAL void duk__bi_print(const char *name, duk__bigint *x) {
 	char buf[DUK__BI_MAX_PARTS * 9 + 64];
 	char *p = buf;
 	duk_small_int_t i;
+  duk_int_t flen;
 
 	/* No NUL term checks in this debug code. */
-	p += DUK_SPRINTF(p, "%p n=%ld", (void *) x, (long) x->n);
+	flen = DUK_SNPRINTF(p, 64, "%p n=%ld", (void *) x, (long) x->n);
+	if (flen > 63) {
+		p += 63;
+	} else if (flen > 0) {
+		p += flen;
+	}
```

Comment:
I had forgotten about the possible negative error return value :o

Negative returns are probably not addressed correctly elsewhere. But it doesn't need to be addressed in this pull.

(Easiest fix would be to have an internal duk_snprintf() helper which would post-process the snprintf() return value: just convert <0 to 0, and cap the return value to bufsize - 1. The return value could then safely be used as is to increment pointers.)